prow/plugins/trickortreat/trickortreat.go

```diff
@@ -0,0 +1,190 @@
+		"https://cdn.pixabay.com/photo/2013/07/12/11/59/candies-145068_640.png",
+		"https://images.pexels.com/photos/4016594/pexels-photo-4016594.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+		"https://i1.pickpik.com/photos/128/614/371/candies-colorful-store-sweet-thumb.jpg",
+		"https://cdn.pixabay.com/photo/2016/07/13/09/43/gummy-bears-1514016_640.png",
+		"https://images.pexels.com/photos/2064126/pexels-photo-2064126.jpeg?auto=compress&cs=tinysrgb&fit=crop&h=627&w=1200",
+		"https://images.pexels.com/photos/2559743/pexels-photo-2559743.jpeg?cs=srgb&dl=awesome-candies-dessert-milkshake-2559743.jpg&fm=jpg",
+		"https://images.pexels.com/photos/4114980/pexels-photo-4114980.jpeg?cs=srgb&dl=candies-and-flowers-on-black-surface-4114980.jpg&fm=jpg",
+		"https://get.pxhere.com/photo/play-sweet-food-color-child-dessert-colour-children-background-bonbons-candy-sweetness-smarties-yummy-sprinkles-confectionery-lentils-snack-food-jelly-bean-1360826.jpg",
+		"https://images.pexels.com/photos/413610/pexels-photo-413610.jpeg?cs=srgb&dl=candies-candy-candy-cane-child-413610.jpg&fm=jpg",
+		"https://get.pxhere.com/photo/group-food-green-red-brown-colorful-yellow-snack-dessert-toy-sugar-bright-multicolored-candy-confection-skittles-assortment-confectionery-candies-jelly-bean-742963.jpg",
+		"https://images.pexels.com/photos/1375807/pexels-photo-1375807.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+		"https://images.pexels.com/photos/1578293/pexels-photo-1578293.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+		"https://images.pexels.com/photos/307280/pexels-photo-307280.jpeg?cs=srgb&dl=cakes-candies-food-sweets-307280.jpg&fm=jpg",
+		"https://images.pexels.com/photos/4015264/pexels-photo-4015264.jpeg?auto=compress&cs=tinysrgb&h=750&w=1260",
+	}
+)
+
+const (
+	pluginName = "trickortreat"
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, _ []config.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The trickortreat plugin adds a candy image to an issue or PR in response to the `/trick-or-treat` command.",
+		Config:      map[string]string{},
+		Snippet:     "",
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/trick(-)or(-)treat",
+		Description: "Add a candy image to the issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Anyone",
+		Examples:    []string{"/trick-or-treat", "/trickortreat"},
+	})
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+}
+
+type snicker interface {
+	readImage() (string, error)
+}
+
+type realSnicker struct {
+}
+
+func (c *realSnicker) readImage() (string, error) {
+	var imgURL string
```

Comment:
This is always nil outside the `for` loop

Reply:
That's a good catch, fixed